packages/fe-mockserver-core/src/mockdata/fileBasedMockData.ts

```diff
@@ -555,10 +555,21 @@ export class FileBasedMockData {
         return isValid;
     }
 
-    async getReferentialConstraints(
-        _navigationProperty: NavigationProperty
-    ): Promise<PartialReferentialConstraint[] | undefined> {
-        return undefined;
+    getReferentialConstraints(_navigationProperty: NavigationProperty): PartialReferentialConstraint[] | undefined {
+        return _navigationProperty.referentialConstraint;
+    }
+    getSourceReference(aggregationAnnotation: RecursiveHierarchy) {
+        const parentNavigationProperty = aggregationAnnotation.ParentNavigationProperty;
+        const referentialConstraint = this.getReferentialConstraints(parentNavigationProperty.$target!)?.[0];
+        if (!parentNavigationProperty.$target) {
+            throw new Error(`Unknown ParentNavigationProperty: '${parentNavigationProperty.value}'`);
+        } else if (!referentialConstraint) {
+            throw new Error(
+                `Referential constraints must be defined for the ParentNavigationProperty: '${parentNavigationProperty.value}'`
+            );
+        }
```

Comment:
something is strange here: `getReferentialConstraints()` accepts a `NavigationProperty` as its parameter --> `parentNavigationProperty.$target!` must be defined.
But then, in the next line: `if (!parentNavigationProperty.$target)...` --> this does not really fit together (only if the target is an empty string), or does it 🤔

Reply:
Yeah the order is stupid :)